### ISSUE TITLE
DSFD: Dual Shot Face Detector

### ISSUE THREAD

**Issue by MariasStory** [justadudewhohacks/face-api.js#274]
Hi, your solution is cool!

What do we need to get following performance?
https://github.com/TencentYoutuResearch/FaceDetection-DSFD
----
DSFD achieves the best performance among all of the state-of-the-art face detectors based on the average precision (AP) across the three subsets, i.e., 96.6% (Easy), 95.7% (Medium) and 90.4% (Hard) on validation set, and 96.0% (Easy), 95.3% (Medium) and 90.0% (Hard) on test set.
----
One problem that I see is the model size:
WIDERFace_DSFD_RES152.pth (459M)
https://drive.google.com/uc?id=1WeXlNYsM6dMP3xQQELI-4gxhwKUQxc3-&export=download

Would be nice to get closer to this one ;-). Even if it takes more memory and time, there might be a use case for it.

Thanks.

**Comment by justadudewhohacks** [comment]
Thanks for posting this! The model indeed is very big in size and complex:

> Figure 2: Our DSFD framework uses a Feature Enhance Module (b) on top of a feedforward VGG/ResNet architecture...

> WIDERFace_DSFD_RES152.pth (459M)

But I will take a look at the paper, sometimes you can apply some of the ideas when training an own model. Also always good to have a reference of state of the art methods and curious how close you can get to state of the art with smaller and faster models, that are actually web friendly.

**Comment by MariasStory** [comment]
Another reference for a good benchmark:
https://github.com/deepinsight/insightface/tree/master/RetinaFace

**Comment by MariasStory** [comment]
The tensorflow implementation:
https://github.com/610265158/DSFD-tensorflow
